src/credential/issuance/06-obtain-credential.ts

```diff
@@ -0,0 +1,141 @@
+  accessToken: Out<AuthorizeAccess>["accessToken"],
+  nonce: Out<AuthorizeAccess>["nonce"],
+  clientId: Out<AuthorizeAccess>["clientId"],
+  credentialType: Out<StartFlow>["credentialType"],
+  context: {
+    credentialCryptoContext: CryptoContext;
+    walletProviderBaseUrl: string;
+    appFetch?: GlobalFetch["fetch"];
+  }
+) => Promise<{ credential: string; format: string }>;
+
+/**
+ * Fetch a credential from the issuer
+ *
+ * @param issuerConf The Issuer configuration
+ * @param accessToken The access token to grant access to the credential, obtained with the access authorization step
+ * @param nonce The nonce value to prevent reply attacks, obtained with the access authorization step
+ * @param clientId Identifies the current client across all the requests of the issuing flow
+ * @param credentialType The type of the credential to be requested
+ * @param context.credentialCryptoContext The context to access the key the Credential will be bound to
+ * @param context.walletProviderBaseUrl The base url of the Wallet Provider
+ * @param context.appFetch (optional) fetch api implementation. Default: built-in fetch
+ * @returns The signed credential token
+ */
+export const obtainCredential: ObtainCredential = async (
+  issuerConf,
+  accessToken,
+  nonce,
+  clientId,
+  credentialType,
+  context
+) => {
+  const {
+    credentialCryptoContext,
+    walletProviderBaseUrl,
+    appFetch = fetch,
+  } = context;
+
+  const credentialUrl = issuerConf.openid_credential_issuer.credential_endpoint;
+
+  const signedDPopForPid = await createDPopToken(
+    {
+      htm: "POST",
+      htu: credentialUrl,
+      jti: `${uuid.v4()}`,
+    },
+    credentialCryptoContext
+  );
+  const signedNonceProof = await createNonceProof(
+    nonce,
+    clientId,
+    walletProviderBaseUrl,
+    credentialCryptoContext
+  );
+
+  const requestBody = {
+    credential_definition: JSON.stringify({
+      type: [credentialType],
+    }),
+    format: "vc+sd-jwt",
+    proof: JSON.stringify({
+      jwt: signedNonceProof,
+      proof_type: "jwt",
+    }),
+  };
+
+  const formBody = new URLSearchParams(requestBody);
+
+  const { credential, format } = await appFetch(credentialUrl, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/x-www-form-urlencoded",
+      DPoP: signedDPopForPid,
+      Authorization: accessToken,
+    },
+    body: formBody.toString(),
+  })
+    .then(hasStatus(200))
+    .then((res) => res.json())
+    .then(CredentialEndpointResponse.parse);
+
+  // validate the received credential signature
+  // is correct and refers to the public keys of the issuer
+  const decodedCredential = await verifyJwt(
+    credential,
+    issuerConf.openid_credential_issuer.jwks.keys
+  ).catch((_) => {
+    // FIXME: [SIW-629] signature verification failures is ignored
+    // as issuer does not implement this feature correctly.
+    // Just logging for now.
+    console.error(_);
```

Comment:
⚠️ I think this is not acceptable. The problem must be fixed by the issuer if it exists

Reply:
Fortunately, it was just a mistake of us. Fixed in https://github.com/pagopa/io-react-native-wallet/pull/62/commits/c5c8c37d35381f3b7d83186018a04cfc4d8f0ff0